lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
+            my $tasks_line_numbers_ref;
+            my $sesrch_ret = ansible_failed_tasks_search_vv($out_f_stdout, $failed_tasks_ref, $tasks_line_numbers_ref);
+            if ($sesrch_ret > 0) {
+                record_info(
+                    "Found failed tasks names",
+                    "Failed tasks unique names ($sesrch_ret):\n" . (join "\n",
+                        @$failed_tasks_ref) .
+                      "\n\nFailed tasks line numbers ($sesrch_ret):\n" . (join "\n",
+                        @$tasks_line_numbers_ref)
+                );
+                my $find_ret = find_ansible_cce_by_task_name_vv($out_ansible_playbook, $failed_tasks_ref, $tasks_line_numbers_ref, $failed_cce_ids_ref, $cce_id_and_name_ref);
+                if ($find_ret > 0) {
+                    record_info(
+                        "Found CCE IDs for failed tasks",
+                        "CCE IDs and tasks names ($find_ret):\n" . (join "\n",
+                            @$cce_id_and_name_ref) . "\n\nCCE IDs ($find_ret):\n" . (join "\n",
+                            @$failed_cce_ids_ref)
+                    );
+                    push(@test_run_report, "failed_cce_ansible_remediation_$remediated = \"" . (join ",",
+                            @$failed_cce_ids_ref) . "\"");
+                }
+                else {
+                    record_info('No failed CCE', "Did not find failed CCE IDs");
+                }
+            }
+            else {
+                record_info('No failed tasks', "Did not find failed tasks");
+            }
         }
-        if ($key && $line =~ m/^Result/) {
-            $parsed{$key} = strip($data[$index + 1]);
-            $key = undef;
+
+        # Upload only stdout logs
+        upload_logs("$f_stdout") if script_run "! [[ -e $f_stdout ]]";
+        upload_logs("$f_stderr") if script_run "! [[ -e $f_stderr ]]";
+    }
+    # If doing bash remediation
+    else {
+        restore_ds_file();
+        my $remediate_cmd = "oscap xccdf eval --profile $profile_ID --remediate --oval-results --report $f_report $f_ssg_ds > $f_stdout 2> $f_stderr";
+        my $ret
+          = script_run("$remediate_cmd", timeout => 3200);
+        record_info("Return=$ret", "$remediate_cmd returns: $ret");
+        if ($ret != 0 and $ret != 2) {
+            record_info('bsc#1194676', 'remediation should be succeeded', result => 'fail');
+            $self->result('fail');
         }
+        # Upload logs & ouputs for reference
+        upload_logs_reports();
     }
-    return %parsed;
+    $remediated++;
+    record_info("Remediated $remediated", "Setting status remediated. Count $remediated");
 }
 
 sub oscap_evaluate {
-    my ($self, $f_ssg_ds, $profile_ID, $n_passed_rules, $n_failed_rules, $eval_match) = @_;
+    # Does evaluation and result analysis
+    my ($self, $eval_match) = @_;
     select_console 'root-console';
 
+    my $n_failed_rules = @$eval_match;
+    my ($failed_rules_ref, $passed_rules_ref);
```

Comment:
feels in general in the code that you are missing intermediate structure to store values, in this case doesn't make sense to save `my` to indicate that are together, anyway are not connected variables. Introductions of hashes with well-named keys could help to read the code and connect parts that are related.